log4j-api/src/main/java/org/apache/logging/log4j/message/MessageFormatMessage.java

```diff
@@ -124,7 +125,7 @@ public boolean equals(final Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (!(o instanceof MessageFormatMessage)) {
```

Comment:
You got an IntelliJ template for this? I'm pretty sure this is generated from the default equals and hashCode templates there.

Reply:
No, I just applied the patches Error Prone suggested (cf. [patching](http://errorprone.info/docs/patching)).